test/polygon/aave/aaveV3-import-test.ts

```diff
@@ -0,0 +1,360 @@
+      type: "function"
+    },
+    {
+      inputs: [
+        { internalType: "address", name: "caller", type: "address" },
+        { internalType: "address", name: "onBehalfOf", type: "address" },
+        { internalType: "uint256", name: "amount", type: "uint256" },
+        { internalType: "uint256", name: "index", type: "uint256" }
+      ],
+      name: "mint",
+      outputs: [{ internalType: "bool", name: "", type: "bool" }],
+      stateMutability: "nonpayable",
+      type: "function"
+    },
+    {
+      inputs: [{ internalType: "address", name: "owner", type: "address" }],
+      name: "nonces",
+      outputs: [{ internalType: "uint256", name: "", type: "uint256" }],
+      stateMutability: "view",
+      type: "function"
+    }
+  ];
+
+  const aaveAbi = [
+    {
+      inputs: [
+        {
+          internalType: "address",
+          name: "asset",
+          type: "address"
+        },
+        {
+          internalType: "uint256",
+          name: "amount",
+          type: "uint256"
+        },
+        {
+          internalType: "uint256",
+          name: "interestRateMode",
+          type: "uint256"
+        },
+        {
+          internalType: "uint16",
+          name: "referralCode",
+          type: "uint16"
+        },
+        {
+          internalType: "address",
+          name: "onBehalfOf",
+          type: "address"
+        }
+      ],
+      name: "borrow",
+      outputs: [],
+      stateMutability: "nonpayable",
+      type: "function"
+    },
+    {
+      inputs: [
+        {
+          internalType: "address",
+          name: "asset",
+          type: "address"
+        },
+        {
+          internalType: "uint256",
+          name: "amount",
+          type: "uint256"
+        },
+        {
+          internalType: "address",
+          name: "onBehalfOf",
+          type: "address"
+        },
+        {
+          internalType: "uint16",
+          name: "referralCode",
+          type: "uint16"
+        }
+      ],
+      name: "deposit",
+      outputs: [],
+      stateMutability: "nonpayable",
+      type: "function"
+    },
+    {
+      inputs: [
+        {
+          internalType: "address",
+          name: "asset",
+          type: "address"
+        },
+        {
+          internalType: "uint256",
+          name: "amount",
+          type: "uint256"
+        },
+        {
+          internalType: "address",
+          name: "onBehalfOf",
+          type: "address"
+        },
+        {
+          internalType: "uint16",
+          name: "referralCode",
+          type: "uint16"
+        }
+      ],
+      name: "supply",
+      outputs: [],
+      stateMutability: "nonpayable",
+      type: "function"
+    }
+  ];
+
+  const daiAbi = abis.basic.erc20;
+
+  let aEth: Contract, aDai: Contract, Dai: any;
+  let dsaWallet0: any;
+  let masterSigner: Signer;
+  let instaConnectorsV2: Contract;
+  let connector: any;
+
+  const wallet = ethers.Wallet.fromMnemonic(mnemonic);
```

Comment:
Why is wallet created like this? We can directly use it, right? Hardhat by default injects the same mnemonic in ethersjs

Reply:
https://github.com/Instadapp/dsa-connectors/blob/main/test/mainnet/compound/compound.test.ts#L25-L26

Reply:
Yes, but to sign the message, we need a private key, which we can get by using the wallet. 

If we use signer to sign a message then the signed message will be having `0x19 Ethereum ...`